src/jvmMain/kotlin/io/beatmaps/api/review.kt

```diff
@@ -492,30 +506,36 @@ fun Route.reviewRoute() {
                 reply.captcha,
                 {
                     val (insertedId, response) = newSuspendedTransaction {
-                        val (mapId, uploadUserId, reviewUserId) = Review
+                        val intermediaryResult = Review
                             .join(Beatmap, JoinType.LEFT, Review.mapId, Beatmap.id)
-                            .select(Review.userId, Beatmap.id, Beatmap.uploader)
+                            .select(Review.userId, Beatmap.id, Beatmap.name, Beatmap.uploader)
                             .where { Review.id eq req.reviewId and Beatmap.deletedAt.isNull() and Review.deletedAt.isNull() }
                             .firstOrNull()
-                            ?.let {
-                                Triple(it[Beatmap.id].value, it[Beatmap.uploader].value, it[Review.userId].value)
-                            } ?: Triple(null, null, null)
 
-                        if (reviewUserId == null || uploadUserId == null || mapId == null) {
+                        if (intermediaryResult == null) {
                             return@newSuspendedTransaction Pair(
                                 null,
                                 ActionResponse(false, listOf("Review or map not found"))
                             )
                         }
 
+                        val reviewUserId = intermediaryResult[Review.userId].value
+                        val mapId = intermediaryResult[Beatmap.id].value
+                        val mapName = intermediaryResult[Beatmap.name]
+                        val uploadUserId = intermediaryResult[Beatmap.uploader].value
+
                         val collaborators = Collaboration
-                            .select(Collaboration.collaboratorId)
+                            .joinCollaborators()
+                            .select(Collaboration.collaboratorId, collaboratorAlias[User.reviewAlerts])
                             .where { Collaboration.mapId eq mapId and Collaboration.accepted }
-                            .map { it[Collaboration.collaboratorId].value }
 
-                        val allowedUsers = listOf(
+                        val mapperIds = listOf(
                             uploadUserId,
-                            *collaborators.toTypedArray(),
+                            *collaborators.map { it[Collaboration.collaboratorId].value }.toTypedArray()
+                        )
+
+                        val allowedUsers = listOf(
+                            *mapperIds.toTypedArray(),
```

Comment:
Is mapperIds used for anything other than composition back into allowedUsers 😕

Reply:
You're right! This PR became a bit more messy than intended.
It used to have another purpose - but it faded out over a few realizations and improvements.
Has been removed now.